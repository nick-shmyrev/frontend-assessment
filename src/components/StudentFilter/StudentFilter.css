.filters {
  position: absolute;
  top: 0;
  left: 20px;
  right: 20px;
  width: auto;
}

input[name="textFilter"] {
  padding: 15px 5px;
  width: 100%;
  border: 0;
  border-bottom: 1px solid #aaa;
  font-size: 24px;
  font-weight: 200;
}
