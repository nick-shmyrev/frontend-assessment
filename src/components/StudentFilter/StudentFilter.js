import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, setTagFilter } from '../../actions/filters-actions';

// Import styles
import style from './StudentFilter.css';

export const StudentFilter = (props) => {
  const onTextFilterChange = (e) => {
    const text = e.target.value;
    props.setTextFilter(text);
  };
  const onTagFilterChange = (e) => {
    const tag = e.target.value;
    props.setTagFilter(tag);
  };
  
  return (
    <div className={style.filters}>
      <input
        name="textFilter"
        type="text"
        placeholder="Search by name"
        value={props.textFilter}
        onChange={onTextFilterChange}
      />
      <input
        name="tagFilter"
        type="text"
        placeholder="Search by tag"
        value={props.tagFilter}
        onChange={onTagFilterChange}
      />
    </div>
  );
};

const mapStateToProps = state => ({
  textFilter: state.filters.text,
  tagFilter: state.filters.tag,
});
const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  setTagFilter: tag => dispatch(setTagFilter(tag)),
});

export default connect(mapStateToProps, mapDispatchToProps)(StudentFilter);
