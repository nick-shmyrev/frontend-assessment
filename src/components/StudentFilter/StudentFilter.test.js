import React from 'react';
import { shallow } from 'enzyme';
import { StudentFilter } from './StudentFilter';

describe('StudentFilter component', () => {
  const filters = {
    text: 'test',
    tag: 'new tag',
  };
  const setTextFilter = jest.fn();
  const setTagFilter = jest.fn();
  
  test('should render correctly', () => {
    const wrapper = shallow(<StudentFilter {...filters} setTextFilter={setTextFilter} setTagFilter={setTagFilter}/>);
    
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should set text filter correctly', () => {
    const value = 'New filter text';
    const wrapper = shallow(<StudentFilter {...filters} setTextFilter={setTextFilter} setTagFilter={setTagFilter}/>);
    
    wrapper.find('input[name="textFilter"]').simulate('change', {
      target: { value },
    });
  
    expect(setTextFilter).toHaveBeenLastCalledWith(value);
  });
  
  test('should set tag filter correctly', () => {
    const value = 'New tag';
    const wrapper = shallow(<StudentFilter {...filters} setTextFilter={setTextFilter} setTagFilter={setTagFilter}/>);
    
    wrapper.find('input[name="tagFilter"]').simulate('change', {
      target: { value },
    });
    
    expect(setTagFilter).toHaveBeenLastCalledWith(value);
  });
});
