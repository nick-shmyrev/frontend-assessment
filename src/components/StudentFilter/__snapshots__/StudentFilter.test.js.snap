// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentFilter component should render correctly 1`] = `
<div
  className="filters"
>
  <input
    name="textFilter"
    onChange={[Function]}
    placeholder="Search by name"
    type="text"
  />
  <input
    name="tagFilter"
    onChange={[Function]}
    placeholder="Search by tag"
    type="text"
  />
</div>
`;
